File: src/main/java/com/backend/app/shared/models/entities/SaleOrder.java
package com.backend.app.shared.models.entities;

import jakarta.annotation.Nullable;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import java.util.Date;


import jakarta.persistence.EnumType;
@Data
@Entity
@Table(name = "sale_orders")
public class SaleOrder  {

    @Id
    @Column(name = "id")
    private String id;

    @Column(name = "order_no")
    private String orderNo;

    @Enumerated(EnumType.STRING)
    @Column(name = "order_status")
    private OrderStatus orderStatus;
    
    @Column(name = "total_quantity")
    private Integer totalQuantity;

    @Column(name = "total_price")
    private Double totalPrice;

    @Column(name = "net_price")
    private Double netPrice;

    @Column(name = "shipping_address")
    @Nullable
    private String shippingAddress;

    @Column(name = "shipping_latitude")
    @Nullable
    private Double shippingLatitude;

    @Column(name = "shipping_longitude")
    @Nullable
    private Double shippingLongitude;

    @Column(name = "street")
    @Nullable
    private String street;

    @Column(name = "sub_district")
    @Nullable
    private String subDistrict;

    @Column(name = "district")
    @Nullable
    private String district;

    @Column(name = "province")
    @Nullable
    private String province;

    @Column(name = "country")
    @Nullable
    private String country;

    @Column(name = "zipcode")
    @Nullable
    private String zipcode;
   
    @Column(name = "created_date")
    private Date createdDate;

    @Column(name = "updated_date")
    private Date updatedDate;

    @Column(name = "note", columnDefinition = "text")
    @Nullable
    private String note;
    
    @Column(name = "customer_firstname")
    private String customerFirstName;
    
    @Column(name = "customer_lastname")
    @Nullable
    private String customerLastName;

    @Column(name = "customer_phone_one")
    @Nullable
    private String customerPhoneOne;

    @Column(name = "customer_phone_two")
    @Nullable
    private String customerPhoneTwo;

}

File: order-svc/models/entities/sale_order.go
package entities

import (
	"time"
)

// OrderStatus is an enum type for the order status field.
// OrderStatus represents the various states an order can have.
type OrderStatus string

const (
	AwaitingPayment     OrderStatus = "AWAITING_PAYMENT"
	AwaitingFulfillment OrderStatus = "AWAITING_FULLFILLMENT"
	Refunded            OrderStatus = "REFUNDED"
	AwaitingShipment    OrderStatus = "AWAITING_SHIPMENT"
	Shipping            OrderStatus = "SHIPPING"
	Completed           OrderStatus = "COMPLETED"
	CancelledByCustomer OrderStatus = "CANCELLED_BY_CUSTOMER"
	CancelledBySystem   OrderStatus = "CANCELLED_BY_SYSTEM"
)

// SaleOrder represents the sale_orders table.
type SaleOrder struct {
	ID                string      `gorm:"column:id;type:varchar(36);primaryKey"`
	OrderNo           string      `gorm:"column:order_no;not null"`
	OrderStatus       OrderStatus `gorm:"column:order_status;type:enum('AWAITING_PAYMENT', 'AWAITING_FULLFILLMENT', 'REFUNDED', 'AWAITING_SHIPMENT', 'SHIPPING', 'COMPLETED', 'CANCELLED_BY_CUSTOMER', 'CANCELLED_BY_SYSTEM');not null"`
	TotalQuantity     int         `gorm:"column:total_quantity;not null"`
	TotalPrice        float64     `gorm:"column:total_price;not null"`
	NetPrice          float64     `gorm:"column:net_price;not null"`
	ShippingAddress   *string     `gorm:"column:shipping_address;type:text"`
	ShippingLatitude  *float64    `gorm:"column:shipping_latitude"`
	ShippingLongitude *float64    `gorm:"column:shipping_longitude"`
	Street            *string     `gorm:"column:street;type:text"`
	SubDistrict       *string     `gorm:"column:sub_district;type:text"`
	District          *string     `gorm:"column:district;type:text"`
	Province          *string     `gorm:"column:province;type:text"`
	Country           *string     `gorm:"column:country;type:text"`
	Zipcode           *string     `gorm:"column:zipcode;type:text"`
	CreatedDate       time.Time   `gorm:"column:created_date;type:timestamp;autoCreateTime"`
	UpdatedDate       time.Time   `gorm:"column:updated_date;type:timestamp;autoUpdateTime"`
	Note              *string     `gorm:"column:note;type:text"`
	CustomerFirstName string      `gorm:"column:customer_firstname;type:text;not null"`
	CustomerLastName  *string     `gorm:"column:customer_lastname;type:text"`
	CustomerPhoneOne  *string     `gorm:"column:customer_phone_one;type:text"`
	CustomerPhoneTwo  *string     `gorm:"column:customer_phone_two;type:text"`
}

// TableName overrides the default table name.
func (SaleOrder) TableName() string {
	return "sale_orders"
}
